lib/datapackage.py

```diff
@@ -9,11 +9,19 @@
 short_names = [script.shortname.lower() for script in SCRIPT_LIST()]
 
 
+def clean_input(prompt, strip=True):
+    val = input(prompt)
+    if strip:
+        val = val.strip()
+    if val == "":
```

Comment:
I could use a loop, but this looks cleaner I suppose.

On Aug 9, 2016 8:16 PM, "Henry Senyondo" <notifications@github.com> wrote:

> In lib/datapackage.py
> <https://github.com/weecology/retriever/pull/599#discussion_r74072635>:
>
> > @@ -9,11 +9,19 @@
> >  short_names = [script.shortname.lower() for script in SCRIPT_LIST()]
> >
> >
> > +def clean_input(prompt, strip=True):
> > +    val = input(prompt)
> > +    if strip:
> > +        val = val.strip()
> > +    if val == "":
>
> why are we using a recursive call?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/weecology/retriever/pull/599/files/5f3d780dac2436dc8aa47e56ef1d534d2469389e..4206dcfc3da0821be50ae7070b2335cd9189de0d#r74072635>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGplhFqKvi3ZmJvlDCHzdtD8BETd46Ezks5qeJKsgaJpZM4Jekew>
> .
>